src/optitrack_node.cpp

```diff
@@ -53,70 +55,147 @@ int main(int argc, char *argv[]) {
+    }else{
+        return -1;
+    }
+
+    //ROS
     ros::Rate loop_rate(240);
-    
-    
     int count = 0;
+    //MAP
+    map<int,ros::Time> frameTimeStamp;
+    bool firstFrameFlag=1;
+    int firstFrameId=0;
+    int localFrameId=0;
+    int localCntFrame=1;
+    int firstHundredFramesCounter=0;
+    static int firstMinCnt;
+    // Create a map iterator and point to beginning of map
+    std::map<int,ros::Time>::iterator MapIterator= frameTimeStamp.begin();
+
+    //FT232
+    std::vector<uint8_t> zero;
+    zero.push_back(0);
+
     while (ros::ok()) {
+        if(ser.isOpen() && (count %2==0))
+        {
+            localFrameId=localCntFrame+firstFrameId;
+            frameTimeStamp[localFrameId]=ros::Time::now();
+            ser.write(zero);
+            localCntFrame++;
+        }
+
         vectorPose poses = mocap.getLatestPoses();
-        ros::Time curTimestamp = ros::Time::now();
-        
-        for(const Pose &curPose : poses){
-            int r = curPose.id - 1;
-            
-            geometry_msgs::Point point;
-            point.x = curPose.t.x();
-            point.y = curPose.t.y();
-            point.z = curPose.t.z();
-
-            geometry_msgs::Quaternion quat;
-            quat.x = curPose.r.x();
-            quat.y = curPose.r.y();
-            quat.z = curPose.r.z();
-            quat.w = curPose.r.w();
-
-            {
-                geometry_msgs::PoseStamped poseStamped;
-                poseStamped.header.frame_id = "optitrack";
-                poseStamped.header.stamp = curTimestamp;
-                poseStamped.header.seq = seqs[r];
-                poseStamped.pose.position = point;
-                poseStamped.pose.orientation = quat;
-
-//            cout << "publishing for " << r << endl;
-//            cout << "cameraMidExposure timestamp = " << curPose.cameraMidExposureTimestamp << endl;
-                rbPubs[r].publish(poseStamped);
-            }
-            {
-                optitrack::RigidBody rigidBody;
-                rigidBody.header.frame_id = "optitrack";
-                rigidBody.header.stamp = curTimestamp;
-                rigidBody.header.seq = seqs[r];
-                rigidBody.pose.position = point;
-                rigidBody.pose.orientation = quat;
-                rigidBody.timestamp = curPose.timestamp;
-                rigidBody.meanError = curPose.meanError;
-                for(const Marker &marker : curPose.markers){
-                    optitrack::Marker markerRos;
-                    markerRos.location.x = marker.location(0);
-                    markerRos.location.y = marker.location(1);
-                    markerRos.location.z = marker.location(2);
-                    markerRos.residual = marker.residual;
-                    markerRos.occluded = marker.occluded;
-                    rigidBody.markers.push_back(markerRos);
+
+        if(firstFrameFlag==0){
+            if(poses[0].FrameNum>0 && frameTimeStamp.count(poses[0].FrameNum)>0 ){
```

Comment:
Czy to się kompiluje poprawnie? Nie powinno być _poses[0].frameNum_

Reply:
Poprawione